library/core/src/async_iter/mod.rs

```diff
@@ -124,5 +124,5 @@
 mod async_iter;
 mod from_iter;
 
-pub use async_iter::{AsyncIterator, IntoAsyncIterator};
+pub use async_iter::{async_iter_next, AsyncIterator, IntoAsyncIterator};
```

Comment:
does this even need to be publically exported?

Reply:
It looks like it does. When I took out the `pub` I got errors in the desugaring about accessing private types.